travis/travis_run_tests

```diff
@@ -31,24 +31,31 @@ def main(test_list):
 
 
 if __name__ == '__main__':
-    do_lint_check = os.environ.get("LINT_CHECK", '1') == "1"
-    do_tests = os.environ.get("TESTS", '1') == "1"
-    export_transifex = os.environ.get("TRANSIFEX") == "1"
-    # Variable TRAVIS_PULL_REQUEST corresponds to pull request number if the
-    # current job is a pull request, "false" if it's not a pull request.
-    travis_pull_request = os.environ.get("TRAVIS_PULL_REQUEST")
+    lint_check_disabled = os.environ.get('LINT_CHECK') == '0'
+    lint_check_enabled = os.environ.get('LINT_CHECK') == '1'
+    tests_enabled = os.environ.get('TESTS') == '1'
+    tests_disabled = os.environ.get('TESTS') == '0'
+    tests_unspecified = os.environ.get('TESTS') is None
+    travis_enabled = os.environ.get('TRANSIFEX') == '1'
+    transifex_enabled = os.environ.get('TRANSIFEX') == '1'
+
+    # TRAVIS_PULL_REQUEST contains the pull request number or 'false'
+    is_pull_request = os.environ.get('TRAVIS_PULL_REQUEST') != 'false'
 
     # Test list. Each test is a list with command + arguments.
     tests = []
-    # By default, do lint checks
-    if do_lint_check:
+
+    if not lint_check_disabled:
         tests.append(['test_flake8'])
         tests.append(['test_pylint'])
-    # Execute tests by default. Disable them on lint checks
-    if not do_lint_check and do_tests:
+
+    if tests_unspecified and not lint_check_enabled:
         tests.append(['test_server.py'])
-    # Execute transifex export with no tests and if it's an official commit
-    if not tests and export_transifex and travis_pull_request == "false":
+
+    if tests_enabled:
+        tests.append(['test_server.py'])
+
+    if transifex_enabled and not tests_disabled and not is_pull_request:
```

Comment:
You're right: since transifex already needs to be enabled explicitly, there is no need to complicate the logic looking into `TESTS`. I fixed that.

As for the comments, I refactored the decisions so that the python code corresponds to the  old comments. At that point they are not useful anymore (and I start to understand something).

Thanks!